benches/store.rs

```diff
@@ -0,0 +1,207 @@
+                            .enumerate()
+                            .map(|(c, v)| (c.try_into().unwrap(), v.into()))
+                            .collect::<Vec<(u64, Word)>>()
+                    },
+                    |l| black_box(MerkleStore::new().with_sparse_merkle_tree(l)),
+                    BatchSize::SmallInput,
+                )
+            },
+        );
+    }
+}
+
+/// Benchmarks getting a node on each of the different storage backends
+///
+/// Since MerkleTree can only be created when a power-of-two number of elements is used, the sample
+/// sizes are limited to that
+fn get_node(c: &mut Criterion) {
+    let mut group = c.benchmark_group("get_node");
+
+    let random_data_size = BATCH_SIZES.into_iter().max().unwrap();
+    let random_data: Vec<RpoDigest> = (0..random_data_size)
+        .map(|_| rand_array::<Felt, 4>().into())
+        .collect();
+
+    for size in BATCH_SIZES {
+        let leaves = &random_data[..size];
+
+        let mtree_leaves: Vec<Word> = leaves.iter().map(|v| v.into()).collect();
+        let mtree = MerkleTree::new(mtree_leaves.clone()).unwrap();
+        let store = MerkleStore::new().with_merkle_tree(mtree_leaves).unwrap();
+        let depth = mtree.depth();
+        let root = mtree.root();
+
+        let smt_leaves = leaves
+            .iter()
+            .enumerate()
+            .map(|(c, v)| (c.try_into().unwrap(), v.into()))
+            .collect::<Vec<(u64, Word)>>();
+        let smt = SimpleSmt::new(63).unwrap().with_leaves(smt_leaves).unwrap();
+
+        group.bench_function(BenchmarkId::new(format!("MerkleTree/{size}"), size), |b| {
+            b.iter_batched(
+                || rand_value::<u64>() % u64::try_from(size).unwrap(),
+                |value| black_box(mtree.get_node(NodeIndex::new(depth, value))),
+                BatchSize::SmallInput,
+            )
+        });
+
+        // This could be done with `bench_with_input`, however to remove variables while comparing
+        // with MerkleTree it is using `iter_batched`
+        group.bench_function(BenchmarkId::new(format!("MerkleStore/{size}"), size), |b| {
+            b.iter_batched(
+                || rand_value::<u64>() % u64::try_from(size).unwrap(),
+                |value| black_box(store.get_node(root, NodeIndex::new(depth, value))),
+                BatchSize::SmallInput,
+            )
+        });
+
+        group.bench_function(BenchmarkId::new(format!("SimpleSmt/{size}"), size), |b| {
+            b.iter_batched(
+                || rand_value::<u64>() % u64::try_from(size).unwrap(),
+                |value| black_box(smt.get_node(&NodeIndex::new(depth, value))),
+                BatchSize::SmallInput,
+            )
+        });
+    }
+}
+
+/// Benchmarks getting a path on each of the different storage backends
+///
+/// Since MerkleTree can only be created when a power-of-two number of elements is used, the sample
+/// sizes are limited to that
+fn get_path(c: &mut Criterion) {
```

Comment:
Similar comment here - I'd have two separate comparisons:

1. `MerkleTree` vs. `MerkleStore` instantiated with `with_merkle_tree`.
2. `SimpleSmt` of depth 63 vs. `MerkleStore` instantiated with `with_sparse_merkle_tree`.

Reply:
split benchmarks